bc_statnice/programove_vypocetni_a_informacni_systemy/01_programovaci_jazyky.md

```diff
@@ -0,0 +1,87 @@
+ 
+# 1. Programovací jazyky
+## Datové struktury programovacích jazyků
+*datová struktura* = způsob uložení informací
+- seznam
+- n-tice
+- množina
+- slovník
+- halda
+- prioritní fronta
+- fronta
+- zásobník
+- vyhledávací strom
+- červeno-černý strom
+- B-strom
+- hašovací tabulka
+
+## Řídící struktury programovacích jazyků
+**C, Java, ...**:
+if ... else, for, while, do ... while, switch
+
+**Python**:
+if...elif...else, while, for
+
+**Haskell**:
+:edit, :load, :quit, :?
+
+**Prolog**:
+repeat...until
+
+## Datové typy
+určují, jakých *hodnot* může objekt nabývat
+- **numerické**: int, float, double, ...
+- **znakové**: char
+- **ukazatelový typ**: \*
+- **definované uživatelem**: struct, union, enum
+určují, jaké *operace* lze s objektem provádět
+
+- **netypované jazyky** - bez typů
+- **slabě typované jazyky** - jazyk přidělí typ při použití -> neočekávané výsledky
+- **silně typované jazyky** - striktní pravidla
+
+**modifikátory datových typů**: signed, unsigned, short, long, long long
+
+## Klasifikace programovacích jazyků:
+- **Imperativní** - jak to naprogramovat (C, Java)
```

Comment:
Za mě spíš něco jako jak to _vypočítat_. Jak to naprogramovat člověk tak nějak řeší vždycky, když programuje :-).

Reply:
:D Taky pravda. Klidně to změň. Původně jsem netušila, že to budu sdílet s dalšími lidmi, takže jsem to psala hlavně tak, abych tomu sama intuitivně rozuměla.